src/edu/psu/sweng500/team8/gui/CombinationCell.java

```diff
@@ -0,0 +1,244 @@
+package edu.psu.sweng500.team8.gui;
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import javax.swing.JToggleButton;
+
+import edu.psu.sweng500.team8.coreDataStructures.Cell;
+import edu.psu.sweng500.team8.play.GameSession;
+
+public class CombinationCell extends javax.swing.JPanel {
+	
+	private final static int PENCIL_MARK_SIZE = 18;
+	private final static int PENCIL_MARk_TEXT_SIZE = 15;
+	private final static Color PENCIL_MARK_COLOR = Color.BLACK;
+	private final static Color PENCIL_MARK_BACKGROUND_COLOR = Color.WHITE;
+	private Cell cell;
+	private GameSession gameSession;
+	
+	public CombinationCell(Cell cell, GameSession gameSession) {
+
+		this.cell = cell;
+		this.gameSession = gameSession;
+		
+		setPreferredSize(new Dimension(PENCIL_MARK_SIZE*3, PENCIL_MARK_SIZE*3));
+		setBorder(null);
+		setAlignmentY(Component.TOP_ALIGNMENT);
+		setAlignmentX(Component.LEFT_ALIGNMENT);
+		GridBagLayout gridBagLayout = new GridBagLayout();
+		setLayout(gridBagLayout);
+		
+		//==
+		JToggleButton tglbtnNewToggleButton_1 = new JToggleButton("1");

```

Comment:
Send me the screen shot.
My design is from Sudoku Pro and I attached is a screenshot.
On the bottom right you can see 3x3 Pencil Mark input mode and next to it
Pencil Mark display mode.

As I stated on the comments, display part is not done and only the input
part is done.
And I am open for any suggestions for display design ideas.


On Tue, Jul 7, 2015 at 7:43 AM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/gui/CombinationCell.java
> <https://github.com/garnryang/Team8/pull/10#discussion_r34030969>:
>
> > +	private GameSession gameSession;
> > +	
> > +	public CombinationCell(Cell cell, GameSession gameSession) {
> > +
> > +		this.cell = cell;
> > +		this.gameSession = gameSession;
> > +		
> > +		setPreferredSize(new Dimension(PENCIL_MARK_SIZE*3, PENCIL_MARK_SIZE*3));
> > +		setBorder(null);
> > +		setAlignmentY(Component.TOP_ALIGNMENT);
> > +		setAlignmentX(Component.LEFT_ALIGNMENT);
> > +		GridBagLayout gridBagLayout = new GridBagLayout();
> > +		setLayout(gridBagLayout);
> > +		
> > +		//==
> > +		JToggleButton tglbtnNewToggleButton_1 = new JToggleButton("1");
>
> I think it is critical that the user can see the pencil marks and the
> entered values at the same time. Otherwise they will have to constantly
> toggle back and forth to see anything, which is a poor design. An example
> is the Sudoku2 App on IOS. Pencil marks are shown in a smaller font in the
> cell. The cells can fit 6 pencil marks each. When a value is entered, the
> pencil marks go away.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/10/files#r34030969>.
>